tasty-query/shared/src/test/scala/tastyquery/TypeSuite.scala

```diff
@@ -376,10 +376,53 @@ class TypeSuite extends UnrestrictedUnpicklingSuite {
     assert(clue(JavaDefinedRef.symbol) == JavaDefinedClass)
   }
 
+  testWithContext("package-private-class") {
+    val javadefinedPackage = ctx.findPackage("javadefined")
+    val PkgPrivateClass = ctx.findTopLevelClass("javadefined.PkgPrivate")
+    val PkgPrivateModule = ctx.findTopLevelModuleClass("javadefined.PkgPrivate")
+    val PkgPrivateModuleClass = ctx.findStaticTerm("javadefined.PkgPrivate")
+
+    def assertPrivateWithin(sym: Symbol, expected: Symbol)(using munit.Location) =
+      assert(!sym.isAnyOf(Flags.Private | Flags.Protected))
+      assert(sym.privateWithin == Some(expected))
+
+    assertPrivateWithin(PkgPrivateClass, javadefinedPackage)
+    assertPrivateWithin(PkgPrivateModule, javadefinedPackage)
+    assertPrivateWithin(PkgPrivateModuleClass, javadefinedPackage)
+  }
+
+  testWithContext("bag-of-java-definitions[static]") {
+    val BagOfJavaDefinitionsClassMod = ctx.findTopLevelModuleClass("javadefined.BagOfJavaDefinitions")
+    val javadefinedPackage = ctx.findPackage("javadefined")
+
+    def testDef(name: TermName)(op: munit.Location ?=> TermSymbol => Unit)(using munit.Location): Unit =
+      op(BagOfJavaDefinitionsClassMod.findNonOverloadedDecl(name))
+
+    testDef(name"STATIC_INT") { STATIC_INT =>
+      assert(STATIC_INT.declaredType.isRef(defn.IntClass))
```

Comment:
Add
```suggestion
      assert(!STATIC_INT.isAnyOf(Flags.Private | Flags.Protected))
      assert(STATIC_INT.privateWithin == None)
```
?

Same for `defaultInt` just below, as well as in the test `"bag-of-java-definitions"`.